### ISSUE TITLE
Problem with rule unknown-less-functions 

### ISSUE THREAD

**Issue by szegi** [racodond/sonar-css-plugin#95]
I started using this plugin and i see that it found an issue
with "extend" function

```
.maine-layout--selected:extend(.maine-layout) {
    color: @text-color-grey;
    .status-new, .status-old, .status-ignore {
        color: @text-color-grey;
    }
}
```

but it doesnt see problem with

```
.client-library__some-data {
    &:extend(.client-list__some-data);
}
```

from what i read, extend is less function from some time
http://lesscss.org/features/#extend-feature

**Comment by racodond** [comment]
Hi,

Which version of the plugin are you running?

David

**Comment by szegi** [comment]
right now version 4.14

**Comment by racodond** [comment]
Hi,

I cannot reproduce the issue.
Could you please provide the entire file as well a screenshot of the issue?

Thank you

David

**Comment by agabrys** [comment]
Hello @racodond,
I wrote a code which allows to reproduce the error:

![unknown-less-functions](https://user-images.githubusercontent.com/10604335/38576636-fd21427e-3cfe-11e8-8b83-25b3f6bf918a.PNG)

The file is compiled successfully.

Environment:
* CSS / SCSS / Less plugin: 4.15
* SonarQube: 7.0

Example file:
https://github.com/agabrys/sonarqube-falsepositives/blob/master/src/main/less/d20180410/unknown-less-functions.less

Project:
https://github.com/agabrys/sonarqube-falsepositives

Build:
mvn clean package sonar

Best Regards

**Comment by racodond** [comment]
Thanks for the reproducer @agabrys!

It's a tricky bug related to at-rule parsing. The violation here is a side-effect from that parsing issue.
keyword: `@media`
prelude:
```
(max-width: 768px) {
    .navtab-info--dates {
        padding-left: 1%;
    }
}

.customization-form__variation-data, .customization-form__variation-menu {
    display: inline-block;
}

.client-library__some-data {
    &:extend(.client-list__some-data);
}

.variation-layout--selected:extend(.variation-layout) {
    color: #000;
    .status-active, .status-inactive, .status-ignore {
        color: #000;
    }
}
```
block: none

instead of 

keyword: `@media`
prelude: `(max-width: 768px)`
block:
```
{
    .navtab-info--dates {
        padding-left: 1%;
    }
}

.customization-form__variation-data, .customization-form__variation-menu {
    display: inline-block;
}

.client-library__some-data {
    &:extend(.client-list__some-data);
}

.variation-layout--selected:extend(.variation-layout) {
    color: #000;
    .status-active, .status-inactive, .status-ignore {
        color: #000;
    }
}
```

**Comment by agabrys** [comment]
I upgraded the plugin to 4.18 and the issue has been marked as fixed.

I'm closing this ticket 😄
